test/cases/api.sh

```diff
@@ -508,21 +510,35 @@ fi
 # generate a temp key for user tests
 ssh-keygen -t rsa-sha2-512 -f /tmp/usertest -C "usertest" -N ""
 
+PAYLOAD_REPOSITORIES_JSON=""
+EXTRA_PACKAGES_JSON=""
+EXTRA_PACKAGE_NAMES=""
+if nvrGreaterOrEqual "osbuild-composer" "41"; then
+  PAYLOAD_REPOSITORIES_JSON=$(cat <<EndOfMessage
+    "payload_repositories": [
+      {
+        "baseurl": "${PAYLOAD_REPO_URL}"
+      }
+    ],
+EndOfMessage
```

Comment:
I don't like the approach of having version conditionals all over our tests. Either skip the api tests entirely in nightly pipelines (the service-facing api isn't supported on-prem anyways). Or somehow install the exact test rpms belonging to the exact nightly version.

Reply:
> Or somehow install the exact test rpms belonging to the exact nightly version.

That's a discussion we've had two months ago and decided to take the current approach to unblock nightly tree testing and make it easier to add to the test suite without the need to go through an approval process and the need to rebase downstream RPMs in order to pick up the new tests.

It's either having to rebase RPMs in order to include new tests or having test conditionals. This PR alone introduces 2 of them. 


I am not in favor of removing tests without the appropriate analysis, even less so delegating this task to a different testing & CI pipeline (IQE). 

The API tests themselves were under debate long time ago but that didn't come to a definitive conclusion. It is a valid point that we need to review these tests and do 2 things: 
- make it clear what is the subject under test (b/c from your comment above it seems like these tests should have never been introduced in the repository or that they are testing multiple items at the same time or both)
-  decide which part goes where and split the work in appropriate chunks.

However this discussion should not be blocking making further progress on the existing nightly pipeline b/c  we risk reaching 8.6 GA without being able to properly test downstream.

Reply:
Is it impossible to get the test rpms which match the exact commit in the nightlies? Because doing it this will make our tests unmaintainable very quickly.

Reply:
> Is it impossible to get the test rpms which match the exact commit in the nightlies? Because doing it like this will make our tests unmaintainable very quickly.

We're trying to solve a different problem here. 

We already have osbuild-composer-tests which is built from the same sources as the osbuild-composer.rpm which goes into nightly trees. That's how things work before this PR. And without this PR we have a chicken-and-egg problem b/c every new test needs to first get into a new version released upstream, then rebased and released downstream and any changes which may be necessary to accommodate the nightly pipeline are usually discovered afterwards and need to go through the standard bugzilla approval process. 

That's why we've taken the approach to disconnect the test suite from the software under test when it comes to running the nightly CI pipelines.